### DISCUSSION TITLE
Add permission bulk selection for non-core collections when setting role permissions

### DISCUSSION THREAD

**Discussion by lluishi93** [directus/directus#18142] [Feature Requests]
### Summary

When creating a new role when Directus already has hundreds of collection, setting all permissions on existing non-core collections takes too much (or has to be done via script).

![image](https://user-images.githubusercontent.com/14786440/231303343-b361ed3e-369e-4ddc-9bc7-e9d6aa157bc9.png)

### Motivation

Fast role creation is the main motivation (and not spending hours setting permissions)

### Detailed Design

Something like checkboxes for CRUD + share permissions that, once are selected and submitted, allows Directus to set those permissions on non-core collections. I can't think of any corner case for this new feature.

### Must Have:
- A button to set all CRUD + share permissions on non-core collections for roles

### Could Have:
- Instead of a button that sets them all, 5 checkboxes for every CRUD operation + Share in order to bulk select specific permissions.

### Drawbacks

I can't think of a drawback for not doing this.

### Alternatives

A sample script to set all those permissions could be a nice alternative.

### Adoption Strategy

It is not a breaking change since this is a new feature that did not even exist in Directus 8.

**Comment by rijkvanzanten** [comment]
Just to confirm, you're talking about a "column" shortcut right? Like enable read access for every collection. Kinda like the row-level `All / None` that already exists currently:

<img width="854" alt="CleanShot 2023-04-11 at 18 42 53@2x" src="https://user-images.githubusercontent.com/9141017/231304517-74a577ac-6238-4868-8f66-4d10fb194048.png">

**Reply by lluishi93** [reply]
Yes. Column shortcut for every non-core collection.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
